modules/howtos/examples/fle.js

```diff
@@ -0,0 +1,167 @@
+"use strict";
+
+const couchbase = require("couchbase");
+const crypto = require("cbfieldcrypt/dist/crypto");
+const aesprovider = require("cbfieldcrypt/dist/aesprovider");
+const keyring = require("cbfieldcrypt/dist/keyring");
+const legacyaesdecrypter = require("cbfieldcrypt/dist/legacyaesdecrypter");
+
+async function oldAnnotation() {
+  // tag::oldannotation[]
+  const mgr = new crypto.DefaultCryptoManager({
+    encryptedFieldPrefix: '__crypt_'
+  })
+  // end::oldannotation[]
+
+  const key = new keyring.Key("mykey", keyB);
+  const insecureKeyring = new keyring.InsecureKeyring(key);
+
+  // tag::legacy[]
+  const decrypter = new legacyaesdecrypter.LegacyAes256Decrypter(insecureKeyring, (publicKey) => {
+    if (publicKey == "mykey") {
+      return "myhmackey"
+    }
+
+    throw Error("unknown key")
+  });
+
+  mgr.registerDecrypter(decrypter);
+  // end::legacy[]
+}
+
+async function go() {
+  const cluster = new couchbase.Cluster("couchbase://localhost", {
+    username: "Administrator",
+    password: "password",
+  });
+
+  const bucket = cluster.bucket("travel-sample")
+  const collection = bucket.scope("tenant_agent_00").collection("users");
+
+  // tag::keys[]
+  const keyB = Buffer.from(
+    '000102030405060708090a0b0c0d0e0f' +
+    '101112131415161718191a1b1c1d1e1f' +
+    '202122232425262728292a2b2c2d2e2f' +
+    '303132333435363738393a3b3c3d3e3f',
+    'hex'
+  ); // 123456789:;<=>?
+
+  const key1 = new keyring.Key("mykey", keyB);
+  const key2 = new keyring.Key("myotherkey", keyB);
+
+  // Create an insecure keyring and add two keys.
+  const insecureKeyring = new keyring.InsecureKeyring(key1, key2);
+  // end::keys[]
+
+  // tag::provider[]
+  // Create a provider.
+  // AES-256 authenticated with HMAC SHA-512. Requires a 64-byte key.
+  const provider = new aesprovider.AeadAes256CbcHmacSha512Provider(insecureKeyring);
+
+  // Create the manager and add the providers.
+  const mgr = new crypto.DefaultCryptoManager();
+
+  // We need to create and then register encrypters.
+  // The key ID here is used by the encrypter to lookup the key from the store when encrypting a document.
+  // The key ID returned from the store at encryption time is written into the data for the field to be encrypted.
+  // The key ID that was written is then used on the decrypt side to find the corresponding key from the store.
```

Comment:
these comments all start with "The key ID" but I think refer to different keys? Could be worth rewording?
Also all passive (not necessarily a problem, except maybe for clarity)